server/common/dao/src/test/java/org/kaaproject/kaa/server/common/dao/service/EndpointSpecificConfigurationServiceImplTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2014-2016 CyberVision, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kaaproject.kaa.server.common.dao.service;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.kaaproject.kaa.common.dto.EndpointProfileDto;
+import org.kaaproject.kaa.common.dto.EndpointSpecificConfigurationDto;
+import org.kaaproject.kaa.server.common.dao.ConfigurationService;
+import org.kaaproject.kaa.server.common.dao.EndpointService;
+import org.kaaproject.kaa.server.common.dao.impl.EndpointSpecificConfigurationDao;
+import org.kaaproject.kaa.server.common.dao.model.EndpointSpecificConfiguration;
+import org.springframework.util.Base64Utils;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+
+public class EndpointSpecificConfigurationServiceImplTest {
+    private static final String KEY = "keyHasg=";
+    private static final String APP_ID = "app";
+    private static final String CONFIG_BODY = "body";
+    private static final byte[] DECODED_KEY = Base64Utils.decodeFromString(KEY);
+    private static final int CONF_VERSION = 7;
+    private static final EndpointSpecificConfigurationServiceImpl SERVICE = new EndpointSpecificConfigurationServiceImpl();
+    private EndpointSpecificConfigurationDao daoMock = mock(EndpointSpecificConfigurationDao.class);
+    private EndpointService endpointServiceMock = mock(EndpointService.class);
+    private ConfigurationService configurationServiceMock = mock(ConfigurationService.class);
+    private EndpointSpecificConfiguration configuration = mock(EndpointSpecificConfiguration.class);
+
+    @Before
+    public void setUp() throws Exception {
+        SERVICE.setEndpointService(endpointServiceMock);
+        SERVICE.setConfigurationService(configurationServiceMock);
+        SERVICE.setEndpointSpecificConfigurationDao(daoMock);
+    }
+
+    @Test
+    public void shouldSave() {
```

Comment:
we name all test methods with test prefix. Although your naming convention may be better, let's not introduce different standards.